textcursor.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gxui
+
+import "github.com/google/gxui/interval"
+
+// TextCursor represents a text position and selection. A selection is defined as
+// the pair of cursor indices sorted least to greatest. When transforming a
+// cursor with no selection, mark should equal index.
+type TextCursor struct {
+	index int
+	mark  int
+}
+
+// NewTextCursor returns a new TextCursor spanning the given indices. If mark equals
+// index, the cursor will have no selection.
+func NewTextCursor(index int, mark int) TextCursor {
+	return TextCursor{index, mark}
+}
+
+// Index returns the cursor index.
+func (c TextCursor) Index() int {
+	return c.index
+}
+
+// Mark returns the cursor mark index.
+func (c TextCursor) Mark() int {
+	return c.mark
+}
+
+// Range returns the difference between indices ordered highest and lowest.
+func (c TextCursor) Range() int {
+	if c.index > c.mark {
+		return c.index - c.mark
+	}
+	return c.mark - c.index
+}
+
+// Selection returns the cursor indices ordered least to greatest.
+func (c TextCursor) Selection() (int, int) {
```

Comment:
`Range()` does make sense if one conflates the idea of a TextCursor and Selection, otherwise if following what I outlined in an earlier comment about the contract the indices a TextCursor follows, some meaning is lost calling this `Range()`